### PR TITLE
Add docs for `copyFile` API

### PR DESCRIPTION
**Pull request by kiview**
This is supposed to re-instate #1989 in a way, that also works on Windows.
So far just minimal changes regarding the original PR, we can probably clean this up a bit more

### CODE REVIEW

File: docs/features/files.md [hunk 1]
```diff
@@ -18,3 +18,19 @@ new GenericContainer(...)
                                       "/etc/redis.conf",
                                       BindMode.READ_ONLY)
 ```
+
+## Copying files to and from containers
```
Comment (bsideup) [inline docs/features/files.md:22]: We should consider putting this section above the one that talks about the filesystem operations

File: docs/features/files.md [hunk 2]
```diff
@@ -18,3 +18,19 @@ new GenericContainer(...)
...
+
+Files can be copied into the container before startup, or can be copied from the container after the container has started.
+
+### Copying to a container before startup
+
+<!--codeinclude-->
+[Copying files to a container](../../core/src/test/java/org/testcontainers/junit/CopyFileToContainerTest.java) inside_block:copyToContainer
```
Comment (bsideup) [inline docs/features/files.md:29]: For some reason the whole file got included:
https://deploy-preview-4661--testcontainers.netlify.app/features/files/

File: docs/features/files.md [hunk 3]
```diff
@@ -18,3 +18,19 @@ new GenericContainer(...)
...
+<!--/codeinclude-->
+
+### Copying a file from a running container
+
+<!--codeinclude-->
+[Copying files from a container](../../core/src/test/java/org/testcontainers/junit/CopyFileToContainerTest.java) inside_block:copyFromContainer
```
Comment (bsideup) [inline docs/features/files.md:35]: For some reason the whole file got included:
https://deploy-preview-4661--testcontainers.netlify.app/features/files/

File: docs/features/files.md (hunk 1 above)
Comment (kiview) [reply docs/features/files.md:22]: Shall we move this section to the top and add a callout box to the other APIs, that they are not supported for every Docker and CI setup?

File: docs/features/files.md (hunk 3 above)
Comment (kiview) [reply docs/features/files.md]: I made it a bit better but did not fix it. Am I misunderstanding how our markdown codeinclude plugin works, or is there some issue with it? Can you have a look at the current effect in this PR @rnorth?

File: docs/features/files.md [hunk 4]
```diff
@@ -11,10 +30,14 @@ new GenericContainer(...)
 
 ## Volume mapping
 
+!!! warning
+    This feature is not supported for all Docker environments. Consider using the `copy` APIs instead.
```
Comment (bsideup) [inline docs/features/files.md:34]: we need to make sure that this does not read as Testcontainers' limitation, but environments'. Maybe mention an example where test host's file system won't match Docker's.

File: docs/features/files.md (hunk 3 above)
Comment (rnorth) [reply docs/features/files.md]: The block name doesn't match:
https://github.com/testcontainers/testcontainers-java/pull/4661/files#diff-54ea1df08fc5691456ef50f20cdaddb8fca38841e7877ebc8ef5d3521cede928R49

`copyFromContainer` vs `copyFileFromContainer`

Comment (kiview) [reply docs/features/files.md]: This is outdated

**Comment by kiview** [comment]
We should extend the docs to also include the new usage introduced by #3815.

**Review by bsideup — APPROVED** [review]

**Pull request merged**